client/keys.go

```diff
@@ -393,6 +393,32 @@ func (k *Key) Quote(selpcr tpm2.PCRSelection, extraData []byte) (*tpmpb.Quote, e
 	return &quote, nil
 }
 
+// Attest generates an Attestation containing the TCG Event Log and a Quote over
+// all PCR banks. The provided nonce can be used to guarantee freshness of the
+// attestation. This function will return an error if the key is not a
+// restricted signing key.
+func (k *Key) Attest(nonce []byte) (*tpmpb.Attestation, error) {
+	sels, err := implementedPCRs(k.rw)
+	if err != nil {
+		return nil, err
+	}
+
+	attestation := tpmpb.Attestation{}
+	if attestation.AkPub, err = k.PublicArea().Encode(); err != nil {
+		return nil, fmt.Errorf("failed to encode public area: %w", err)
+	}
+	attestation.Quotes = make([]*tpmpb.Quote, len(sels))
+	for i, sel := range sels {
+		if attestation.Quotes[i], err = k.Quote(sel, nonce); err != nil {
```

Comment:
What's the reason to quote each pcr separately? Can we just quote the sels one time?

Reply:
This is a limitation of [`QuoteRaw`](https://pkg.go.dev/github.com/google/go-tpm@v0.3.2/tpm2#QuoteRaw), it can only take a single [`PCRSelection`](https://pkg.go.dev/github.com/google/go-tpm@v0.3.2/tpm2#PCRSelection) despite the underlying command supporting a list of selections.

Having separate quotes also helps with quote verification (so one PCR set getting messed up doesn't ruin the whole thing).